extensions/interactions/GraphInput/directives/graph-viz.component.spec.ts

```diff
@@ -0,0 +1,1881 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the graph-viz.
+ */
+
+import { async, ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { GraphVizComponent } from './graph-viz.component';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { MockTranslatePipe } from 'tests/unit-test-utils';
+import { GraphDetailService } from './graph-detail.service';
+import { DeviceInfoService } from 'services/contextual/device-info.service';
+import { PlayerPositionService } from 'pages/exploration-player-page/services/player-position.service';
+import { EventEmitter } from '@angular/core';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+
+describe('GraphVizComponent', () => {
+  let component: GraphVizComponent;
+  let graphDetailService: GraphDetailService;
+  let deviceInfoService: DeviceInfoService;
+  let playerPositionService: PlayerPositionService;
+  let focusManagerService: FocusManagerService;
+  let fixture: ComponentFixture<GraphVizComponent>;
+
+  let mockNewCardAvailableEmitter = new EventEmitter();
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [GraphVizComponent, MockTranslatePipe],
+      providers: [GraphDetailService, DeviceInfoService],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    graphDetailService = TestBed.inject(GraphDetailService);
+    deviceInfoService = TestBed.inject(DeviceInfoService);
+    playerPositionService = TestBed.get(PlayerPositionService);
+    focusManagerService = TestBed.inject(FocusManagerService);
+    fixture = TestBed.createComponent(GraphVizComponent);
+    component = fixture.componentInstance;
+
+    component.graph = {
+      vertices: [
+        {
+          x: 150,
+          y: 50,
+          label: ''
+        },
+        {
+          x: 200,
+          y: 50,
+          label: ''
+        },
+        {
+          x: 150,
+          y: 100,
+          label: ''
+        },
+        {
+          x: 196.60939025878906,
+          y: 95.05902099609375,
+          label: ''
+        }
+      ],
+      isDirected: false,
+      isWeighted: true,
+      isLabeled: false,
+      edges: [
+        {
+          src: 0,
+          weight: 1,
+          dst: 1
+        },
+        {
+          src: 1,
+          weight: 1,
+          dst: 2
+        }
+      ]
+    };
+
+    component.canAddVertex = true;
+    component.canDeleteVertex = true;
+    component.canMoveVertex = true;
+    component.canEditVertexLabel = true;
+    component.canAddEdge = true;
+    component.canDeleteEdge = true;
+    component.canEditEdgeWeight = true;
+    component.interactionIsActive = true;
+    component.canEditOptions = true;
+  });
+
+  it('should initialise component, when a graph is visualized', () => {
```

Comment:
The test description does not exactly portray what is happening inside the test block (`the graph is visualized part`). Can you modify the description or add a comment?